src/hidimstat/marginal/selection_variable_scikit_learn.py

```diff
@@ -0,0 +1,302 @@
+        importances for the given data.
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Input data matrix.
+        y : array-like of shape (n_samples,)
+            Target values.
+        Raises
+        ------
+        NotImplementedError
+            This method should be implemented by subclasses.
+        """
+        raise NotImplementedError()
+
+    def fit_importance(self, X, y, cv=None):
+        """
+        Fit the model and compute feature importances.
+        This method fits the model (if necessary) and computes feature
+        importances for the given data.
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Input data matrix.
+        y : array-like of shape (n_samples,)
+            Target values.
+        cv : None or int, optional
+            (not used) Cross-validation parameter.
+        Returns
+        -------
+        importances_ : ndarray
+            Feature importance scores.
+        """
+        if cv is not None:
+            warnings.warn("cv won't be used")
+        self.fit()
+        return self.importance(X, y)
+
+
+class AnalysisOfVariance(AdapterScikitLearn):
+    """
+    Analysis of Variance (ANOVA) :footcite:t:`fisher1970statistical` feature
+    selection for classification tasks.
+    This class uses scikit-learn's f_classif to compute F-statistics and p-values
+    for each feature. For a short summary of this method, see
+    :footcite:t:`larson2008analysis`.
+    Attributes
+    ----------
+    importances_ : ndarray
+        1 - p-values for each feature (higher is more important).
+    pvalues_ : ndarray
+        1 - p-values for each feature.
+    f_statitstic_ : ndarray
+        F-statistics for each feature.
+    Notes
+    -----
+    See sklearn.feature_selection.f_classif
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    @override
+    def importance(self, X, y):
+        """
+        Compute ANOVA F-statistics and p-values for each feature.
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Input data matrix.
+        y : array-like of shape (n_samples,)
+            Target class labels.
+        Sets
+        ----
+        importances_ : ndarray
+            1 - p-values for each feature.
+        pvalues_ : ndarray
+            1 - p-values for each feature.
+        f_statitstic_ : ndarray
+            F-statistics for each feature.
+        Returns
+        -------
+        importances_ : ndarray
+            1 - p-values for each feature.
+        """
+        f_statistic, p_values = f_classif(X, y)
+        # Test the opposite hypothese to the anova
+        # Test the similarity in the distribution instead of the difference
+        self.importances_ = 1 - p_values
+        self.pvalues_ = 1 - p_values
+        self.f_statitstic_ = f_statistic
```

Comment:
```suggestion
        self.importances_ = f_statistic
        self.pvalues_ = p_values
```
Similar to the discussion for D0CRT keeping the test statistics as importance score avoids redundancy an is more intuitive. 
self.p_values_ however should correspond to the p_values

Reply:
The p-value from scikitlearn tests is the opposite question; it's why it's important to take 1 - p_values.

f_static can't be interpreted as importances_ because it has a different meaning. It's the additional statistical test for the difference in variance of the distribution.

Reply:
But precisely a variance difference  _is_ a good measure of variable importance. This is actually the basis for Total Sobol index, and other more recent measures that we use for complex learners. When you take a loss difference, you're essentially doing that. We could argue about the distinction between variance difference/variance ratio and the actual F values, but it is OK to take the F as an importance measure.  It's in any case better than a p-value.

Reply:
```python
import numpy as np
from sklearn.feature_selection import f_classif

X = np.random.randn(200, 5)
y = (X[:, 0] + X[:, 1]) > 0

f_stat, pval = f_classif(X, y)
for i in range(X.shape[1]):
    print(f"Feature {i}: F-stat: {f_stat[i]}, p-val: {pval[i]}")
```

Reply:
Feature 0: F-stat: 75.79396233015946, p-val: 1.2340835313704089e-15
Feature 1: F-stat: 104.22807168051357, p-val: 6.264844580007097e-20
Feature 2: F-stat: 0.1573650119634395, p-val: 0.6920213870884148
Feature 3: F-stat: 0.7371543590566706, p-val: 0.3916121439370208
Feature 4: F-stat: 0.7699702415100195, p-val: 0.3812897221335523

As you can see, the p-value is close to zeros for important values.

Reply:
Yes, this is the expected result for the test $X^j \perp\perp Y | X^{-j}$

Reply:
For me, this is not the same test that we want. We want to know if $X^{j}$ is dependent on $Y$ and not if it's independent.

What do you think?